private/buf/buflsp/completion.go

```diff
@@ -106,13 +106,68 @@ func completionItemsForDeclPath(ctx context.Context, file *file, declPath []ast.
 }
 
 // completionItemsForSyntax returns the completion items for the files syntax.
-func completionItemsForSyntax(ctx context.Context, file *file, syntaxDecl ast.DeclSyntax) []protocol.CompletionItem {
+func completionItemsForSyntax(ctx context.Context, file *file, syntaxDecl ast.DeclSyntax, position protocol.Position) []protocol.CompletionItem {
 	file.lsp.logger.DebugContext(ctx, "completion: syntax declaration", slog.Bool("is_edition", syntaxDecl.IsEdition()))
 
-	var prefix string
+	positionLocation := file.file.InverseLocation(int(position.Line)+1, int(position.Character)+1, positionalEncoding)
+	offset := positionLocation.Offset
+
+	valueSpan := syntaxDecl.Value().Span()
+	start, end := valueSpan.Start, valueSpan.End
+	valueText := valueSpan.Text()
+
+	// Break on newlines to split an unintended capture.
+	if index := strings.IndexByte(valueText, '\n'); index != -1 {
```

Comment:
This is needed on `syntax = "_` where theres no closing `"` the token reports the whole file as the value capture. Must split. This is the same logic as import statements so we can probably factor this bit out too.